docs/UserGuide.md

```diff
@@ -73,14 +109,46 @@ VirusTracker would be able to generate useful information based off the data tha
-  e.g. `deleteLocation 3` deletes the 3rd location shown in the location list while `deleteLocation idl/L123` deletes the location with Id `L123`.
-
 * **Data** refers collectively to people, locations and visits unless stated otherwise.
+
+* VirusTracker will actively stop the user from creating duplicates of entities. VirusTracker will alert you if it detects you are adding a duplicate entity.
+  Entities are duplicates if they meet any of the following conditions.
+  1. They have the same id.
+  2. For visits, they are duplicates if they have the same date and involve the same person and location.
+  3. For persons, they are duplicates if their name is the same and one of phone and email is the same.
+  4. For locations, they are duplicates if they have the same name and address. 
+</div>
+
+There are many situations where you may want to refer to a specific location or person when giving a command.
+For example, you may want to delete a location or add a visit involving a specific location and person.
+In such cases, VirusTracker provides two methods for you to use when referring to a specific person or location, either index or id.
+This guide collectively refers to indexes and ids as identifiers.
+In the figure below showing a person panel, there is a number beside the person's name and there is a person id.<br>
+ ![personPanel](images/personPanel.png)
+
+You may use either the index number (which corresponds to the position of the person within the persons list) or the id to refer to a person. 
+The same applies for locations.
+In the example shown above, you may use either `deletePerson 1` or `deletePerson idp/S123A` to remove the person named `Alex Yeoh` from VirusTracker.
+You may prefer to use ids when they do not wish to scroll the list to find the relevant index.
+You may also prefer to use indexes which tend to be shorter than the ids of people and locations.
+
+<div markdown="block" class="alert alert-info">
+
+**:information_source: Notes about using indexes and ids:**<br>
+* The field `IDENTIFIER` means that the user needs to input either an id or index.
+  `LOCATION_IDENTIFIER` and `PERSON_IDENTIFIER` refers to location and person identifiers respectively.
+* You may input an index by just using the number alone. However, ids must have a prefix in front of them.
+  `idp` is the prefix for person while `idl` is the prefix for locations. <br>
+  e.g.  `deleteLocation 3` and `deleteLocation idl/L123A`. The first command uses an index of 3 while the second command uses the id `L123A`.
+* When using indexes, it should before any other fields which need prefixes. <br>
+  e.g. `addVisit 1 1 d/2020-02-02` is allowed but `addVisit 1 d/2020-02-02 1` is not allowed.
+* Take note that the index of a person/location may change depending on the index of the person/location as viewed from the most recently displayed person/location list.
```

Comment:
This section is a bit information packed, perhaps you could consider adding small diagrams illustrating your points? For example, for the part about index changing, you could show an example of the same person but with a different index after say, an add operation.

Reply:
I have chosen to separate the information into separate lists based on their content. One list is for how to use indexes and ids and the other is for warnings to take note about. I believe this will help improve readability.